src/components/tl/dpu/tl_dpu_team.c

```diff
@@ -0,0 +1,216 @@
+/**
+ * Copyright (C) Mellanox Technologies Ltd. 2021.  ALL RIGHTS RESERVED.
+ *
+ * See file LICENSE for terms.
+ */
+
+#include "tl_dpu.h"
+#include "tl_dpu_coll.h"
+#include "coll_score/ucc_coll_score.h"
+
+UCC_CLASS_INIT_FUNC(ucc_tl_dpu_team_t, ucc_base_context_t *tl_context,
+                    const ucc_base_team_params_t *params)
+{
+    ucc_tl_dpu_context_t *ctx = ucc_derived_of(tl_context, ucc_tl_dpu_context_t);
+
+    UCC_CLASS_CALL_SUPER_INIT(ucc_tl_team_t, &ctx->super);
+
+    ucp_mem_map_params_t mmap_params;
+    ucp_request_param_t send_req_param, recv_req_param;
+    tl_dpu_ucp_request_t *send_req[3], *recv_req[2];
+    size_t rem_rkeys_lengths[3];
+    uint64_t rem_addresses[3];
+    void *ctrl_seg_rkey_buf;
+    size_t ctrl_seg_rkey_buf_size;
+    size_t total_rkey_size;
+    void *rem_rkeys;
+
+    self->coll_id   = 1;
+    self->size      = params->params.oob.participants;
+    self->rank      = params->rank;
+
+    mmap_params.field_mask = UCP_MEM_MAP_PARAM_FIELD_ADDRESS |
+                             UCP_MEM_MAP_PARAM_FIELD_LENGTH;
+    mmap_params.address    = (void*)&self->ctrl_seg;
+    mmap_params.length     = sizeof(self->ctrl_seg);
+
+    ucp_mem_map(ctx->ucp_context, &mmap_params, &self->ctrl_seg_memh);
+    ucp_rkey_pack(ctx->ucp_context, self->ctrl_seg_memh,
```

Comment:
check ret codes

Reply:
done - also added a few more ret code checks through out the code